spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -852,6 +900,119 @@ def create_obj_in_recycled_slot(should_sample_original: false)
     end
   end
 
+  describe '#stats' do
+    it 'returns basic lifetime stats of stack recorder' do
+      num_serializations = 5
+
+      num_serializations.times do
+        stack_recorder.serialize
+      end
+
+      stats = stack_recorder.stats
+
+      expect(stats).to match(
+        hash_including(
+          :serialization_successes => num_serializations,
+          :serialization_failures => 0,
+
+          :serialization_time_ns_min => be > 0,
+          :serialization_time_ns_max => be > 0,
+          :serialization_time_ns_avg => be > 0,
+          :serialization_time_ns_total => be > 0,
+
+          :heap_recorder_snapshot => nil,
+        )
+      )
+
+      serialization_time_min, serialization_time_max, serialization_time_avg, serialization_time_total =
+        stats.values_at(
+          :serialization_time_ns_min,
+          :serialization_time_ns_max,
+          :serialization_time_ns_avg,
+          :serialization_time_ns_total
+        )
+
+      expect(serialization_time_min).to be <= serialization_time_avg
+      expect(serialization_time_avg).to be <= serialization_time_max
+      expect(serialization_time_total).to eq serialization_time_avg * num_serializations
+    end
+
+    context 'with heap profiling enabled' do
+      let(:heap_samples_enabled) { true }
+      let(:heap_size_enabled) { true }
+
+      before do
+        skip 'Heap profiling is only supported on Ruby >= 2.7' if RUBY_VERSION < '2.7'
+      end
+
+      after do |example|
+        # This is here to facilitate troubleshooting when this test fails. Otherwise
+        # it's very hard to understand what may be happening.
+        if example.exception
+          puts('Heap recorder debugging info:')
+          puts(described_class::Testing._native_debug_heap_recorder(stack_recorder))
+        end
+      end
+
+      def sample_allocation(obj)
+        # Heap sampling currently requires this 2-step process to first pass data about the allocated object...
+        described_class::Testing._native_track_object(stack_recorder, obj, 1, obj.class.name)
+        Datadog::Profiling::Collectors::Stack::Testing
+          ._native_sample(Thread.current, stack_recorder, { 'alloc-samples' => 1 }, [], [], 400, false)
+      end
+
+      it 'includes heap recorder snapshot' do
+        live_objects = []
+        live_heap_samples = 6
+        live_heap_samples.times do |i|
+          obj = "nice living string #{i}"
+          obj.freeze if i.odd? # Freeze every other string
```

Comment:
Using strings may not be a great idea here as people may start sprinkling `#frozen_string_literal` on files and break this. Perhaps a `Object.new` instead?